pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -211,12 +239,42 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 		c.ExtraTags = strings.Split(instance.ExtraTags, ",")
 	}
 
-	if instance.Network != "" {
-		log.Warnf("`network_address` config is not available for corecheck SNMP integration to use autodiscovery. Agent `snmp_listener` config can be used instead: https://docs.datadoghq.com/network_monitoring/devices/setup?tab=snmpv2#autodiscovery")
+	c.Network = instance.Network
+	if instance.DiscoveryWorkers == 0 {
+		c.DiscoveryWorkers = defaultDiscoveryWorkers
+	} else {
+		c.DiscoveryWorkers = instance.DiscoveryWorkers
+	}
```

Comment:
You can simplify logic with
```suggestion
	c.DiscoveryWorkers = defaultDiscoveryWorkers
	if instance.DiscoveryWorkers != 0 {
		c.DiscoveryWorkers = instance.DiscoveryWorkers
	}
```

Reply:
> You can simplify logic with

Not sure one is less/more simplified than the other, just different.

It's more a matter of preference, I would prefer not changing it if you are ok.

Reply:
I see it similar to an early return - if I were to refactor this to a `getDiscoveryWorker` I would use
```
if (instance.DiscoveryWorker== 0) {
  return defaultDiscoveryWorkers
}
return instance.DiscoveryWorkers
```

Anyhow I can live with one extra line of code.